contrib/epee/include/readline_buffer.h

```diff
@@ -19,12 +21,23 @@ namespace rdln
     }
     void get_line(std::string& line) const;
     void set_prompt(const std::string& prompt);
+    static void add_completion(const std::string& command)
+    {
+      if(std::find(completion_commands.begin(), completion_commands.end(), command) != completion_commands.end())
+	return;
+      completion_commands.push_back(command);
```

Comment:
>> Now that I expanded the code I see that this is actually duplicating an existing std::map
> Where?

I assume you are referring to the private map in `command_handler`? For such a small vector to exist in the readline code I really don't see the issue of such minimal duplication. It feels preferable than to hold a reference to console_handler and adding a public getter to that map.